### DISCUSSION TITLE
WYSIWYG UX Revamp

### DISCUSSION THREAD

**Discussion by zcolah** [zesty-io/manager-ui#2308] [Ideas]
# We would love to hear your feedback on how we can improve the WYSIWYG UX 
We are beginning to explore and define how to improve the user experience of the WYSIWYG editor inside of the Content application of Zesty. A primary focus of this revamp  is to make multiple UX enhancements based on feedback from our community, and hence we would like to start this conversation with all of you on how our WYSIWYG editor can improve. 

### **Wish to share your feedback over a video call with our team? [Schedule a call with me today](https://calendly.com/zesty-reviews/design-review)!** 

# Design Explorations
Here are some initial design explorations based on the research findings of our team!

## Re-ordering Buttons to make it easier to find insert image etc.
We have re-ordered the buttons to make the experience more intuitive and consistent with other rich text editors in the market. Key changes include:

1. Making the Text Type (Heading 1) as the first option. Block Quotes and Code Type can also now be found in this dropdown. 
2. Bringing the Insert features such as Insert Image, Video, URL, etc to the top bar
3. Full Screen has been shifted to the top bar 
4. Code Icon Button is now used for Embed which is consistent with other experiences. Instead of clicking the code button, to edit the HTML of the WYSIWYG output, the user will now click on the HTML button.

We would love to hear your opinion on how useful the new order of the buttons is as compared to the old experience? 

![image](https://github.com/zesty-io/manager-ui/assets/22625919/893d0c80-05e9-4785-9299-5bfed8b67d28)

## Slash Commands
Similar to Notion, Coda, and other apps out there we are considering introducing a Slash command. By using the slash “/” key a drop down appears, with elements like tables and images listed, plus heading. See screenshot ideas below.  

- We would like to here how useful this feature would be? 
- Which design idea do you prefer - Idea 1 or Idea 2? 
- Do you have any feedback on what the ordering of the items should be in the menu? 

**Idea 1 - Single Column Menu with no suggestions**
![image](https://github.com/zesty-io/manager-ui/assets/22625919/9b04cd3c-b9c4-47a6-a957-aa5677a291fa)

**Idea 2 - Multi Column Menu with suggestions** 
![image](https://github.com/zesty-io/manager-ui/assets/22625919/200b613a-094f-4024-b2ad-e59d42963da7)

## Floating Text Formatting Toolbar
Simply highlight a block of text you wish to format and use the floating toolbar to format the text further. See screenshot below. 

On a scale of 1 (not useful) to 5 (super useful), how useful would you find this feature? How many times a week do you see yourself using a feature such as this? 

![image](https://github.com/zesty-io/manager-ui/assets/22625919/62e83242-2da1-41aa-84ff-6865c225c895)

## Full Screen Mode
Previously the text content would stretch the entire width of the screen in full screen making it super difficult to read. With the new experience, the text content width is constrained for easy reading and along with a document outline being provided for easy scanning of content. 

- We would love to know your opinion on the width used? Should we increase or decrease it? 
- On a scale of 1 (not useful) to 5 (super useful), how useful would you find the document outline feature? How many times a week do you see yourself using a feature such as this? 

![image](https://github.com/zesty-io/manager-ui/assets/22625919/da343939-7941-4979-9d6a-a4d45c42fb59)

## Users Can Now Preview Images and Copy Image URLs directly from the WYSIWYG
Hovering over an image in a WYSIWYG, now reveals the file name along with an option to copy the file url and preview the file which when clicked opens the full media preview of the file.
Note: Users can use the arrow keys in the Media preview to toggle between the images in the WYSIWYG 

![image](https://github.com/zesty-io/manager-ui/assets/22625919/efdceeb1-9e3b-4f11-bf48-d27c3d1dc817)

# Other Problems and Areas of improvement we have discovered: 

1. **Keeping the WYSIWYG toolbar should be fixed to the top of the screen**
2. **Make the WYSIWYG Field have a reduced Min Height and Max Height.** This is because we heard the real estate the WYSIWYG takes is a lot. For example, If it was really used only for a small paragraph it does not need to take so much space. It should wrap to the content in it and have a max height. 
3. **Add menu that triggers when a user is pasting content into the WYSIWYG text input that asks them if they want to paste as plain text or rich text.**
4. **Restrict media uploads within a WYSIWYG field to a specific folder**
5. **Upload Image to Media when a user drags and drops an image into the WYSIWYG upload it to Media**
6. **Provide an error when a user tries to drag and drop a non image file into the WYSIWYG editor**
7. **Use New Tiny MCE Version (need to check what the different between our version and the new version is)**

**Comment by ardeay** [comment]
Thank you for this research @zcolah 

Two image requests from my end:

- The ability to "paste" a screenshot or copied image from clipboard directly
- Ability to drag an image from my desktop into the WYSIWYG editor
